.github/workflows/format.yml

```diff
@@ -6,26 +6,21 @@ jobs:
   prettier:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.ref }}
       - name: Install dependencies
         run: npm ci
-      - name: Cache node_modules
-        uses: actions/cache@v3
-        with:
-          path: |
-            ./node_modules
-          key: npm-workspace-cache-${{ runner.os }}-${{ hashFiles('./package-lock.json') }}
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v3
         with:
           node-version: 16
+          cache: 'npm'
       # Workflows are not allowed to edit workflows. As result, we need to prevent Prettier from formatting them.
       - name: Prevent workflows from being formatted
         run: echo ".github" >> .prettierignore
       - run: npm run fmt
       # Prevent the prettierignore change from being committed.
       - run: git checkout .prettierignore
-      - uses: stefanzweifel/git-auto-commit-action@v4
+      - uses: stefanzweifel/git-auto-commit-action@8756aa072ef5b4a080af5dc8fef36c5d586e521d # tag=v5
```

Comment:
I assume you did a security review on these. Is it worth sticking a comment in here to document that?

Reply:
Checking in a tag + pinned SHA like this is exactly comparable to checking in a version range + lockfile for an npm dependency - why require comments here but not there?